opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/MonitoringLocationRestServiceIT.java

```diff
@@ -164,4 +177,208 @@ public void testEventOnCreationAndDeletion() throws Exception {
+        // 15 > default, should return 10
+        assertEquals(10, testLocationLimit(15,DEFAULT_FLAG));
+    }
+    @Test
+    @Transactional
+    public void testUnlimitedLocations5() throws Exception {
+        // 5 <= unlimited, should return 5 (includes default)
+        assertEquals(5, testLocationLimit(5,UNLIMITED_FLAG));
+    }
+    @Test
+    @Transactional
+    public void testUnlimitedLocations10() throws Exception {
+        // 10 <= unlimited, should 10
+        assertEquals(10, testLocationLimit(10,UNLIMITED_FLAG));
+    }
+    @Test
+    @Transactional
+    public void testUnlimitedLocations15() throws Exception {
+        // 15 <= unlimited, should return 15
+        assertEquals(15, testLocationLimit(15,UNLIMITED_FLAG));
+    }
+    @Test
+    @Transactional
+    public void testLimit5Locations5() throws Exception {
+        // 5 <= specified limit of 5, should return 5
+        assertEquals(5, testLocationLimit(5,5));
+    }
+
+    @Test
+    @Transactional
+    public void testLimit5Locations10() throws Exception {
+        // 10 > specified limit of 5, should return 5
+        assertEquals(5, testLocationLimit(10,5));
+    }
+    @Test
+    @Transactional
+    public void testLimit5Locations15() throws Exception {
+        // 15 > specified limit of 5, should return 5
+        assertEquals(5, testLocationLimit(15,5));
+    }
+    @Test
+    @Transactional
+    public void testLimit10Locations5() throws Exception {
+        // 5 <= specified limit of 10, should return 6 (includes default)
+        assertEquals(5, testLocationLimit(5,10));
+    }
+
+    @Test
+    @Transactional
+    public void testLimit10Locations10() throws Exception {
+        // 10 = specified limit of 10, should return 10
+        assertEquals(10, testLocationLimit(10,10));
+    }
+
+    @Test
+    @Transactional
+    public void testLimit10Locations15() throws Exception {
+        // 10 > specified limit of 10, should return 10
+        assertEquals(10, testLocationLimit(15,10));
+    }
+
+    @Test
+    @Transactional
+    public void testLimit15Locations5() throws Exception {
+        // 5 <= 15 specified limit of 15, should return 5 (includes default)
+        assertEquals(5, testLocationLimit(5,15));
+    }
+    @Test
+    @Transactional
+    public void testLimit15Locations10() throws Exception {
+        // 10 <= specified limit of 15, should return 10
+        assertEquals(10, testLocationLimit(10,15));
+    }
+    @Test
+    @Transactional
+    public void testLimit15Locations15() throws Exception {
+        // 10 > specified limit of 15, should return 15
+        assertEquals(15, testLocationLimit(15,15));
+    }
+
+    public static Integer calcLimit(final Integer flaggedLimit){
+        if (flaggedLimit == DEFAULT_FLAG) {
+            return DEFAULT_LIMIT;
+        } else if (flaggedLimit.equals(UNLIMITED_FLAG)){
+            return UNLIMITED;
+        } else {
+            return flaggedLimit;
+        }
+    }
```

Comment:
I really do not understand why use use "FLAGS" instead of using just the integer values. We have just a few cases:
1. location count < default value, e.g. count 5 < limit 10 -> results 5
2. location count > default value, e.g. count 11 > limit 10 -> results 10
3. limit manually set to a custom value, e.g. count 10 > limit 6 -> results 6
4. limit manually set to zero, e.g. count 11 > limit 0 -> results 11

Reply:
Should I cater for cases when DEFAULT_LIMIT might change?

Reply:
In the case the DEFAULT_LIMIT changes, the test must be changed to the new DEFAULT_LIMIT.

Reply:
If there is at least on test having more the default limit elements the test will fail as soon as the limit is changed. Which is fine here.

Reply:
Ok